client/components/ReviewsWidget.jsx

```diff
@@ -165,11 +199,89 @@ export class ReviewsWidget extends Component {
   *************************/
 
   handleSortChange(event) {
+    // Resets relevant parts of state to reset the review list view
+    // based off of the newly selected sort
     let sortOrder = event.target.value;
     let page = stateDefaults.page;
     let numToDisplay = stateDefaults.numToDisplay;
     let productReviews = [];
-    this.setState({ sortOrder, page, numToDisplay, productReviews });
+    let allReviewsFetched = stateDefaults.allReviewsFetched;
+    this.setState({ sortOrder, page, numToDisplay, productReviews, allReviewsFetched });
+  }
+
+  /******************************
+   * ===== MARK HELPFUL ======= *
+  ******************************/
+
+  markReviewHelpful(reviewId, index) {
+    let prefix = 'review';
+
+    // Check browser storage to see if this user has marked this
+    // review as helpful before.
+    let reviewsMarked = localStorage.getItem(prefix + reviewId);
+
+    // If they have not previously marked this review
+    if (!reviewsMarked) {
+
+      this.apiMarkHelpful(reviewId)
+      .then(()=>{
+      // Update hepfulness number in local filtered review list
+      let filteredReviews = this.state.filteredReviews.slice();
+      let review = filteredReviews[index];
+      review.helpfulness += 1;
+      review = Object.assign({}, review); // Copy so React knows it changed
+      filteredReviews[index] = review;
+
+      localStorage.setItem(prefix + reviewId, true);
+      this.setState({
+        filteredReviews
+      })
+    })
```

Comment:
Are you updating the whole collection of reviews every time someone clicks "helpful"?